msm/src/serialization/constraints.rs

```diff
@@ -185,3 +185,71 @@ impl<F: PrimeField> InterpreterEnv<F> for Env<F> {
         }))
     }
 }
+
+impl<Fp: PrimeField> Env<Fp> {
+    #[allow(dead_code)]
+    // FIXME: not mut
+    fn constrain_lookups(&mut self) -> Vec<E<Fp>> {
+        assert_eq!(self.rangecheck4_lookups.len(), 20);
+        assert_eq!(self.rangecheck15_lookups.len(), 17);
+
+        {
+            let rc4_t_lookup = Lookup {
+                table_id: LookupTable::RangeCheck4,
+                numerator: curr_cell(Column::LookupMultiplicity(
```

Comment:
I am confused...

If I had understood correctly, you need a multiplicity for each table entry, not table id. No?

Reply:
It is for the expression `m(X)/(\beta + t(X)`

Reply:
Explained offline